### PR TITLE
Add notebook with examples for Codepencence by Marti

### PR DESCRIPTION
**Pull request by PanPip**
This notebook provides explanations of the GPR and the GNPR distances that allow discriminating random variables both on distribution and dependence. These metrics are described in the work of Gautier Marti "Some contributions to the clustering of financial time series and applications to credit default swaps" and implemented in the mlfinlab package.

### CODE REVIEW

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb [hunk 1]
```diff
@@ -0,0 +1,1200 @@
+{
+ "cells": [
```
Comment (Jackal08) [inline Codependence/Codependence by Marti/codependence_by_marti.ipynb:2]: Lets add an abstract at the top.

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb [hunk 2]
```diff
@@ -0,0 +1,1200 @@
...
+  {
+   "cell_type": "markdown",
```
Comment (Jackal08) [inline Codependence/Codependence by Marti/codependence_by_marti.ipynb:4]: Your heatmaps need titles & please use `;` or `plt.show()` to remove the `<matplotlib.axes._subplots.AxesSubplot at 0x227c5590588>` text in your plots.

Comment (Jackal08) [reply Codependence/Codependence by Marti/codependence_by_marti.ipynb:4]: Maybe also add a label to the heat bar on the right ?

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb [hunk 3]
```diff
@@ -0,0 +1,1200 @@
...
+   "metadata": {},
+   "source": [
+    "* By: Illya Barziy\n",
```
Comment (Jackal08) [inline Codependence/Codependence by Marti/codependence_by_marti.ipynb:7]: Add a references section at the bottom of the notebook and make of the Harvard referencing method.

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb [hunk 4]
```diff
@@ -0,0 +1,1200 @@
...
+    "* Email: illyabarziy@gmail.com\n",
+    "* Reference: __Some contributions to the clustering of financial time series and applications to credit default swaps__ _by_ Gautier Marti"
```
Comment (Jackal08) [inline Codependence/Codependence by Marti/codependence_by_marti.ipynb:9]: Ruifan's notebooks set a good example for some of the layout, abstract, references, labels.

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb (hunk 1 above)
Comment (PanPip) [reply Codependence/Codependence by Marti/codependence_by_marti.ipynb:2]: Done

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb (hunk 2 above)
Comment (PanPip) [reply Codependence/Codependence by Marti/codependence_by_marti.ipynb:4]: Done, also increased the size of the heatmaps

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb (hunk 3 above)
Comment (PanPip) [reply Codependence/Codependence by Marti/codependence_by_marti.ipynb:7]: Done

File: Codependence/Codependence by Marti/codependence_by_marti.ipynb (hunk 4 above)
Comment (PanPip) [reply Codependence/Codependence by Marti/codependence_by_marti.ipynb:9]: Separated sections of the notebook using that style

**Pull request merged**